sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -549,15 +549,18 @@ class HDFSFileCatalog(
 
   refresh()
 
+  private def isInvalidFile(f: FileStatus): Boolean =
+    f.getPath.getName.startsWith("_") || f.getLen == 0
```

Comment:
Yeah, that assert is my fault (from when I though my bin packing algorithm had a bug).  We should support reading empty files (if thats valid for the format) and we should avoid writing them when possible.